File: Python/Cpu.py
from Komponen import Komponen

class Cpu(Komponen):
    def __init__(self, jumlah_core=0, kecepatan_ghz=0.0, socket="", merk="", nama="", harga=0):
        super().__init__(merk, nama, harga)
        self.jumlah_core = jumlah_core
        self.kecepatan_ghz = kecepatan_ghz
        self.socket = socket
        
    def set_jumlah_core(self, jumlah_core):
        self.jumlah_core = jumlah_core
        
    def set_kecepatan_ghz(self, kecepatan_ghz):
        self.kecepatan_ghz = kecepatan_ghz
        
    def set_socket(self, socket):
        self.socket = socket
        
    def get_jumlah_core(self):
        return self.jumlah_core
        
    def get_kecepatan_ghz(self):
        return self.kecepatan_ghz
        
    def get_socket(self):
        return self.socket
        
    def display_info(self):
        super().display_info()
        print(f"Jumlah Core: {self.jumlah_core}")
        print(f"Kecepatan: {self.kecepatan_ghz} GHz")
        print(f"Socket: {self.socket}")

File: Java/Cpu.java
public class Cpu extends Komponen {
    private int jumlahCore;
    private float kecepatanGHz;
    private String socket;

    public Cpu() {
        super();
        this.jumlahCore = 0;
        this.kecepatanGHz = 0.0f;
        this.socket = "";
    }

    public Cpu(int jumlahCore, float kecepatanGHz, String socket, String merk, String nama, int harga) {
        super(merk, nama, harga);
        this.jumlahCore = jumlahCore;
        this.kecepatanGHz = kecepatanGHz;
        this.socket = socket;
    }

    public void setJumlahCore(int jumlahCore) {
        this.jumlahCore = jumlahCore;
    }

    public void setKecepatanGHz(float kecepatanGHz) {
        this.kecepatanGHz = kecepatanGHz;
    }

    public void setSocket(String socket) {
        this.socket = socket;
    }

    public int getJumlahCore() {
        return this.jumlahCore;
    }

    public float getKecepatanGHz() {
        return this.kecepatanGHz;
    }

    public String getSocket() {
        return this.socket;
    }

    @Override
    public void displayInfo() {
        super.displayInfo();
        System.out.println("Jumlah Core: " + this.jumlahCore);
        System.out.println("Kecepatan: " + this.kecepatanGHz + " GHz");
        System.out.println("Socket: " + this.socket);
    }
}
